### ISSUE TITLE
Support dwt for libinput

### ISSUE THREAD

**Issue by ModProg** [k0kubun/xremap#152]
This is not really a bug of xremap I think, but a configuration problem.

# Problem

With xremap activated, `dwt` libinputs "disable-while-typing" is not working.

# Reproduce

Run `xremap` on a laptop and enable `dwt` (e.g. for sway, you do that by adding `input * dwt enabled` to your config).

While typing, try to move your mouse with the touchpad, if it was working correctly your touchpad should not be controlling your mouse (this should also be the case without `xremap` running).

On my device, the touchpad disabling while typing did not work when `xremap` is running, but did work when I killed `xremap`.

# Proposed solution
Following https://gitlab.freedesktop.org/libinput/libinput/-/issues/524 the solution would be to convince `libinput` that the `xremap` keyboard is an internal keyboard if the source is.

Maybe this requires manual intervention to just put something similar to Depau in the `/etc/libinput/local-overrides.quirks`:
```ini
[xremap]
MatchUdevType=keyboard
MatchBus=usb //or whatever we need
MatchName= // whatever xremap uses
AttrKeyboardIntegration=internal
```

**Comment by ModProg** [comment]
I'm using it without `MatchBus` and `MatchName` i.e. with all keyboards, as I never use my laptop with an external keyboard and it works as expected.

**Comment by ModProg** [comment]
> Sorry, I didn't get what you're trying to solve. Can you possibly rearrange your issue description with a few headers, such as "Background", "Problem", "Proposed solution"? I feel like you skipped some of them or made them a bit cryptic.

sorry,

**Comment by k0kubun** [comment]
Could you also describe how to reproduce the problem? I don't know much about `dwt` and `disable-while-typing`.

**Comment by ModProg** [comment]
> Could you also have another section to describe how to reproduce the problem? I don't know much about `dwt` and `disable-while-typing`.

It's a feature to disable the touchpad on a laptop while typing.

**Comment by k0kubun** [comment]
Okay, thanks for the clarifications. As I don't have a Linux laptop, I wouldn't be able to reproduce it locally. For now, as I need to focus on conference preparations for a few weeks, please allow me to have a look at this after Sep 10th or so. I could look at a patch if you have a fix on the xremap side in your mind.

**Comment by ModProg** [comment]
That's alright, I have the before mentioned workaround for now, but am also not sure how libinput detects whether a keyboard is an internal or an external keyboard.

**Comment by ModProg** [comment]
I asked for how to do this without a manually configuring as the user on the libinput gitlab: https://gitlab.freedesktop.org/libinput/libinput/-/issues/801

**Comment by k0kubun** [comment]
It looks like you've got an answer at https://gitlab.freedesktop.org/libinput/libinput/-/issues/801. Because I have no Linux laptop, there's no way for me to reproduce the problem and fix it. Can you file a pull request if it's something fixable on the xremap's end?

Alternatively, if it's not something an xremap binary can deal with, maybe we could start with documenting your workaround in your README for now. In the future, we could incorporate the configuration in xremap packaging, which I can't work on soon, but I've thought about it. For that, what OS distribution do you use?

**Comment by ModProg** [comment]
I'm using Arch. I think I have an idea on how to implement this dynamically and will implement this, hopefully in the near future.

**Comment by rytswd** [comment]
@ModProg Sorry to resurrect an old ticket , but I'm seeing the same behaviour discussed here, where having Xremap running would cause `dwt` to be ignored. Is there anything else I need to configure for Xremap or my user setup? I'm using 0.10.0 version, and installed on NixOS with flake.

**Comment by ModProg** [comment]
@rytswd does the work around work for you? https://github.com/xremap/xremap/issues/152#issuecomment-1227097914 work for you?

**Comment by rytswd** [comment]
@ModProg I haven't tried a workaround, thinking that #196 was meant to fix it -- do I have to get the workaround in place? I'm on NixOS, so I need to dig into how I can add the config in Nix way.
Just so that I understand your point, would you be able to share the exact config you have? Is it just the `[xremap]` definition in the issue description, without MatchBus and MatchName?

**Comment by ModProg** [comment]
@rytswd #196 should have fixed it, I was just proposing to check if the workaround would make any difference.

IIRC, it worked like this when I used the workaround. (For me, #196 did fix the issue)
```ini
[xremap]
MatchUdevType=keyboard
AttrKeyboardIntegration=internal
```

**Comment by rytswd** [comment]
@ModProg Thanks for the info! As I'm on NixOS, the libinput config is quite different, and I tried a few options but in the end couldn't figure out where to add that code snippet in place... I'll do a bit more digging later this week.
Considering that it's working fine with you, I suspect I'm missing something  in my setup. I had some misconfiguration earlier with `uinput` missing (which I got support with https://github.com/xremap/nix-flake/issues/70), so I may need to open up a ticket there to seek support from NixOS users.
